File: javascript/beer-song/beer-song.js
//
// This is only a SKELETON file for the 'Beer Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const lyrics = [];
  
  for (let i = 0; i < takeDownCount; i++) {
    const currentBottles = initialBottlesCount - i;
    
    // Add an empty line between verses (except for the first verse)
    if (i > 0) {
      lyrics.push('');
    }
    
    // First line of the verse
    if (currentBottles === 0) {
      lyrics.push('No more bottles of beer on the wall, no more bottles of beer.');
    } else {
      lyrics.push(`${formatBottles(currentBottles)} of beer on the wall, ${formatBottles(currentBottles)} of beer.`);
    }
    
    // Second line of the verse
    if (currentBottles === 0) {
      // Special case for 0 bottles
      lyrics.push('Go to the store and buy some more, 99 bottles of beer on the wall.');
    } else {
      const nextBottles = currentBottles - 1;
      const takeAction = currentBottles === 1 ? 'Take it down' : 'Take one down';
      const nextBottlesText = nextBottles === 0 ? 'no more bottles' : formatBottles(nextBottles);
      lyrics.push(`${takeAction} and pass it around, ${nextBottlesText} of beer on the wall.`);
    }
  }
  
  return lyrics;
};

// Helper function to format the bottle count with proper grammar
function formatBottles(count) {
  if (count === 0) {
    return 'No more bottles';
  } else if (count === 1) {
    return '1 bottle';
  } else {
    return `${count} bottles`;
  }
}

File: python/beer-song/beer_song.py
def recite(start, take=1):
    """
    Recite the lyrics to the "99 Bottles of Beer on the Wall" song.
    
    Args:
        start: The starting number of bottles
        take: The number of verses to recite (default: 1)
    
    Returns:
        A list of strings representing the lyrics
    """
    lyrics = []
    
    for bottles in range(start, start - take, -1):
        # First line
        if bottles == 0:
            lyrics.append("No more bottles of beer on the wall, no more bottles of beer.")
        elif bottles == 1:
            lyrics.append("1 bottle of beer on the wall, 1 bottle of beer.")
        else:
            lyrics.append(f"{bottles} bottles of beer on the wall, {bottles} bottles of beer.")
        
        # Second line
        if bottles == 0:
            lyrics.append("Go to the store and buy some more, 99 bottles of beer on the wall.")
        elif bottles == 1:
            lyrics.append("Take it down and pass it around, no more bottles of beer on the wall.")
        elif bottles == 2:
            lyrics.append("Take one down and pass it around, 1 bottle of beer on the wall.")
        else:
            lyrics.append(f"Take one down and pass it around, {bottles - 1} bottles of beer on the wall.")
        
        # Add empty line between verses, but not after the last verse
        if bottles > start - take + 1:
            lyrics.append("")
    
    return lyrics
